public/app/features/explore/AddToDashboard/AddToDashboardModal.tsx

```diff
@@ -24,17 +24,17 @@ function withRedirect<T extends any[]>(fn: (redirect: boolean, ...args: T) => {}
 }
 
 export const AddToDashboardModal = ({ onClose, queries, visualization, onSave }: Props) => {
-  const [submissionError, setSubmissionError] = useState<string>();
   const {
     register,
     handleSubmit,
     control,
     formState: { errors, isSubmitting },
     setError,
-  } = useForm<FormDTO>({ defaultValues: { queries, visualization } });
+    clearErrors,
+    // fakeField is used to track non-specific-input-bound form errors.
+  } = useForm<FormDTO & { fakeField: unknown }>({ defaultValues: { queries, visualization } });
```

Comment:
@gelicia  `fakeField` may also be renamed to something different, like `formError` or anything else. it just blurs a bit the line between what is form state (ie. validation status) and operation state. the idea for NOT going with this approach is to separate between actionable API errors (for example validation, when we ask the user to change something in order to make data valid) and non-actionable errors (ie. something went wrong, we don't know why).